src/app/dashboard/freelancer/page.tsx

```diff
@@ -57,25 +58,57 @@ interface Project {
 export default function Dashboard() {
   const user = useSelector((state: RootState) => state.user);
   const [projects, setProjects] = useState<Project[]>([]);
-  const [loading, setLoading] = useState(true);
+  const [activeProjects, setActiveProjects] = useState<Project[]>([]);
+  const [pendingProjects, setPendingProjects] = useState<Project[]>([]);
   const [showMeetingDialog, setShowMeetingDialog] = useState(false); // State for showing dialog
+  const [currentTab, setCurrentTab] = useState('Active');
+  const [loading, setLoading] = useState(true);
+  const [loadingStats, setLoadingStats] = useState(true);
+
+  const fetchData = async (status: string) => {
+    setLoading(true);
+    try {
+      const response = await axiosInstance.get(
+        `/freelancer/${user.uid}/project?status=${status}`,
+      );
+      setProjects(response.data.data);
+    } catch (error) {
+      console.error('API Error:', error);
+    } finally {
+      setLoading(false);
+    }
+  }; // Fetch data when selectedStatus changes
 
+  const fetchProjectStats = async () => {
+    setLoadingStats(true);
+    try {
+      const activeCountResponse = await axiosInstance.get(
+        `/freelancer/${user.uid}/project?status=Active`,
+      );
+      const pendingCountResponse = await axiosInstance.get(
+        `/freelancer/${user.uid}/project?status=Pending`,
+      );
+
+      setActiveProjects(activeCountResponse.data.data);
+      setPendingProjects(pendingCountResponse.data.data);
+    } catch (error) {
+      console.error('API Error for project stats:', error);
+    } finally {
+      setLoadingStats(false);
+    }
+  };
+
+  /* eslint-disable react-hooks/exhaustive-deps */
   useEffect(() => {
-    const fetchData = async () => {
-      try {
-        const response = await axiosInstance.get(
-          `/freelancer/${user.uid}/project`,
-        );
-        setProjects(response.data.data);
-      } catch (error) {
-        console.error('API Error:', error);
-      } finally {
-        setLoading(false);
-      }
-    };
+    fetchProjectStats();
+    fetchData(currentTab);
+  }, [user.uid, currentTab]);
```

Comment:
Here changing status will trigger this use effect and call both data fetch function, which is unnecessary 
Think of a better approach for fetching data